src/theory/rewriter.cpp

```diff
@@ -149,6 +149,32 @@ TheoryRewriter* Rewriter::getTheoryRewriter(theory::TheoryId theoryId)
   return d_theoryRewriters[theoryId];
 }
 
+Node Rewriter::rewriteViaRule(ProofRewriteRule id, const Node& n)
+{
+  // dispatches to the appropriate theory
+  TheoryId tid = theoryOf(n);
```

Comment:
Note: it may be better to find the theory rewriter from the `ProofRewriteRule` instead of the node.

Reply:
That is true, although `theoryOf` isn't that much more expensive than a map lookup.  This way is slightly more elegant since it requires a bit less bookkeeping.